### ISSUE TITLE
[false-positive] disconnected if-statements in foreach loop with variables assigned by reference

### ISSUE THREAD

**Issue by Azenet** [kalessil/phpinspectionsea#794]
An if statement will be marked as disconnected from its parent foreach if the variable it is checking is assigned by reference and has not been set before.
```php
foreach ($item as $i) {
    passthru($i, $return);

    if ($return > 0) { // This statement seems to be disconnected from its parent foreach.
        ...
    }
}
```

**Comment by kalessil** [comment]
Opps, thank you for reporting @Azenet

**Comment by funivan** [comment]
Don`t know if this is the same issue. But anyway here is one more false-positive
```php
<?php
  declare(strict_types=1);
  $subject = '';
  foreach ([1, 2, 3, 4] as $index) {
    list($matched, $used) = [null, null];   // <-- this statement could be decooupled
    if ($index % 2 === 0) {
      preg_match('!\d+!', '', $matched);
    }
    echo is_array($matched) ? 'ok' : 'nok';
    echo "\n";
  }
```
If I move `list($matched, $used) = [null, null];` one line up - I will get wrong code.

**Comment by kalessil** [comment]
@funivan : got it (another case, but I'll fix it as well).

**Comment by kalessil** [comment]
@Azenet : your case has been fixed, thank you for reporting =)

**Comment by Azenet** [comment]
@kalessil Thank you!

**Comment by kalessil** [comment]
@Azenet : you are welcome =)

@funivan : your case fixed as well.
